Assets/BossRoom/Scripts/Shared/Game/GameEventMessage.cs

```diff
@@ -0,0 +1,26 @@
+namespace Unity.Multiplayer.Samples.BossRoom
+{
+    public struct LifeStateChangedEventMessage
+    {
+        public LifeState NewLifeState;
+        public FixedPlayerName CharacterName;
+        public CharacterTypeEnum CharacterType;
+
+        public static int Size => sizeof(LifeState) + FixedPlayerName.Size + sizeof(CharacterTypeEnum);
```

Comment:
you don't need to manually calculate this, you can just do sizeof(MyStruct). Since there's no polymorphism (that I can see?), everything in here should be known at compile time

Reply:
sizeof requires an unsafe context for everything except for primitive types (https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/sizeof), so from my understanding, that's why I cannot apply it directly on the struct and  get compile errors if I try.

Reply:
You can look at what FastBufferWriter does with its GetWriteSize method
        public static unsafe int GetWriteSize<T>() where T : unmanaged
This is what we use for our own internal messages (for example ServerRpcMessage)

Reply:
Also note you can enable unsafe code at the assembly level (doesn't have to be for the whole project). It's a checkbox in the assembly's options

Reply:
as discussed, a cleaner solution was integrated in #605